### PR TITLE
Implement organization dao

### PR DESCRIPTION
**Pull request by mperehinets**

### CODE REVIEW

File: src/main/java/com/smartcity/dao/OrganizationDaoImpl.java [hunk 1]
```diff
@@ -1,22 +1,118 @@
...
+            jdbcTemplate.update(connection -> {
+                PreparedStatement ps = connection.prepareStatement(Queries.SQL_ORGANIZATION_CREATE,
+                        Statement.RETURN_GENERATED_KEYS);
+                ps.setString(1, organization.getName());
+                ps.setString(2, organization.getAddress());
+                ps.setObject(3, organization.getCreatedDate());
+                ps.setObject(4, organization.getUpdatedDate());
+                return ps;
+            }, holder);
+            organization.setId(Optional.ofNullable(holder.getKey().longValue())
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/dao/OrganizationDaoImpl.java]: holder.getKey() may return null so you will get NPE, implement this way:
`Optional.ofNullable(holder.getKey()).map(Number::longValue)
                    .orElseThrow(() -> new DbOperationException("Create organization error: AutoGeneratedKey = null"))`

File: src/main/java/com/smartcity/domain/Organization.java [hunk 2]
```diff
@@ -1,5 +1,92 @@
...
+    private String name;
+    private String address;
+    private LocalDateTime  createdDate;
+    private LocalDateTime updatedDate;
+
+    public Organization() {
+
+    }
+
+    public Organization(Long id, String name, String address, LocalDateTime created_date, LocalDateTime updated_date) {
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/domain/Organization.java]: Naming convention

File: src/main/java/com/smartcity/domain/Organization.java [hunk 3]
```diff
@@ -1,5 +1,92 @@
...
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public LocalDateTime getCreatedDate() {
+        return createdDate;
+    }
+
+    public void setCreatedDate(LocalDateTime created_date) {
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/domain/Organization.java]: Naming convention

File: src/main/java/com/smartcity/domain/Organization.java [hunk 4]
```diff
@@ -1,5 +1,92 @@
...
+        this.updatedDate = updated_date;
+    }
+
+    @Override
+    public String toString() {
+        return "Organization{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                ", address='" + address + '\'' +
+                ", created_date=" + createdDate +
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/domain/Organization.java]: Naming convention

File: src/main/java/com/smartcity/mapper/OrganizationMapper.java [hunk 5]
```diff
@@ -2,12 +2,29 @@
...
+        return instance;
+    }
+
     public Organization mapRow(ResultSet resultSet, int i) throws SQLException {
-        return null;
+        Organization organization = new Organization();
+        organization.setId(resultSet.getLong("id"));
+        organization.setName(resultSet.getString("name"));
+        organization.setAddress(resultSet.getString("address"));
+        organization.setCreatedDate(resultSet.getTimestamp("created_date").toLocalDateTime());
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/mapper/OrganizationMapper.java]: Naming convention

File: src/test/java/com/smartcity/dao/OrganizationDaoImplTest.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    public static void start() {
+        setup();
+        organizationDao = new OrganizationDaoImpl(dataSource);
+    }
+
+    @Test
+    public void testCreateOrganization() {
+        organizationDao.create(organization);
+        assertThat(organizationDao.create(organization)).isEqualToIgnoringGivenFields(organization,
+                "created_date", "updated_date");
```
Comment (iholovatyi) [inline src/test/java/com/smartcity/dao/OrganizationDaoImplTest.java]: Naming convention

File: src/main/java/com/smartcity/mapper/OrganizationMapper.java [hunk 7]
```diff
@@ -2,12 +2,29 @@
...
+    }
+
+    public static OrganizationMapper getInstance() {
+        return instance;
+    }
+
     public Organization mapRow(ResultSet resultSet, int i) throws SQLException {
-        return null;
+        Organization organization = new Organization();
+        organization.setId(resultSet.getLong("id"));
```
Comment (iholovatyi) [inline src/main/java/com/smartcity/mapper/OrganizationMapper.java]: May restultSet be null? If so then handle it properly

File: src/main/java/com/smartcity/dao/OrganizationDaoImpl.java (hunk 1 above)
Comment (mperehinets) [reply src/main/java/com/smartcity/dao/OrganizationDaoImpl.java]: ok )

File: src/main/java/com/smartcity/domain/Organization.java (hunk 2 above)
Comment (mperehinets) [reply src/main/java/com/smartcity/domain/Organization.java]: ok )

File: src/main/java/com/smartcity/domain/Organization.java (hunk 3 above)
Comment (mperehinets) [reply src/main/java/com/smartcity/domain/Organization.java]: ok )

File: src/main/java/com/smartcity/mapper/OrganizationMapper.java (hunk 5 above)
Comment (mperehinets) [reply src/main/java/com/smartcity/mapper/OrganizationMapper.java]: This is the name of the table from the database

File: src/test/java/com/smartcity/dao/OrganizationDaoImplTest.java (hunk 6 above)
Comment (mperehinets) [reply src/test/java/com/smartcity/dao/OrganizationDaoImplTest.java]: ok )

File: src/main/java/com/smartcity/mapper/OrganizationMapper.java (hunk 7 above)
Comment (mperehinets) [reply src/main/java/com/smartcity/mapper/OrganizationMapper.java]: ok )

File: src/main/java/com/smartcity/mapper/OrganizationMapper.java (hunk 5 above)
Comment (iholovatyi) [reply src/main/java/com/smartcity/mapper/OrganizationMapper.java]: makes sense

**Comment by iholovatyi** [comment]
@mperehinets squash all your commits into one

**Pull request closed without merging**